app/src/main/java/com/navapp/navigation/util/math/TSP/HelpKarpSolver.java

```diff
@@ -0,0 +1,112 @@
+package com.navapp.navigation.util.math.TSP;
+
+import com.navapp.navigation.util.math.MathLocation;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+//This class is based on the code from the Heuristics-TSP library by Sinclert Pérez, @Sinclert on Github, see Heuristics-TSP-license.md
+public class HelpKarpSolver<T extends MathLocation> implements TSPSolver<T> {
+    private double optimalCost;
+    private List<T> optimalPath;
+
+    public HelpKarpSolver() {
+        resetVars();
+    }
+
+    @Override
+    public Queue<T> solve(T start, List<T> locations, T end) {
```

Comment:
Where are the tests... for everything? As far as we know this might not even work.